### PR TITLE
MRG: ECoG channel type

### LINKED ISSUE #2661: Change 'seeg' to 'ieeg', or add an 'ecog' channel type
I noticed that there is an 'seeg' electrode type allowed for many functions. I wonder if it was intentional to use 'seeg' to specifically mean depth electrodes, or if this is more of an umbrella term to refer to intracranial electrodes (which would include both depth electrodes and surface electrodes).

What are thoughts about either:

A. Changing 'seeg' to 'ieeg', so that it's a catchall term for intracranial recordings.
B. Keeping it that way but adding an 'ecog' type, so that it's explicit which one you're dealing with.

It's just not clear to me why stereo eeg would be added but not surface electrodes.

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #2661.

Ready for review/merge from my end.

@choldgraf can you try it out a bit see if it works decently? I'm especially curious if the values chosen for plotting make sense.

### CODE REVIEW

**Comment by choldgraf** [comment]
Sounds good - is there any test data for ECoG yet? Or you want me to just try it w/ my own? It might be tricky to have a single value for ECoG because there are so many recording setups, but I'll try it on some data sets from a few different sites and see if there are differences.

**Comment by Eric89GXL** [comment]
Just try on some of your own. We don't have any native ECoG data format readers coded, so going through *Array class is all we can probably do now anyway (and it's what's done in the tests I wrote).

**Comment by choldgraf** [comment]
Cool... I often get ecog data in EDF format actually. Though that's still
highly site dependent

Sent from my phone, apologies for the curtness and type-os
On Apr 22, 2016 9:58 AM, "Eric Larson" <notifications@github.com> wrote:

Just try on some of your own. We don't have any native ECoG data format
readers coded, so going through *Array class is all we can probably do now
anyway (and it's what's done in the tests I wrote).

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/mne-tools/mne-python/pull/3157#issuecomment-213512205>

**Comment by Eric89GXL** [comment]
> I often get ecog data in EDF format actually

If you can get some tiny file, we can add support to our EDF reader

**Comment by choldgraf** [comment]
I think you're missing some scalings information in the docstrings (e.g., plot_raw and accompanying method for the Raw object). When I plotted some data the scalings were too high, and my lines looked flat. But again this could be a site-specific thing. I'm trying to track down some "totally raw" data that hasn't been processed at all.

**Comment by Eric89GXL** [comment]
Can you try playing with these values in `mne/defaults.py` to try to get something reasonable? Hopefully that's enough to get you going. In particular, `scalings=dict(..., seeg=1e4, ..., ecog=1e6)` right now, so maybe `ecog` needs to be 1e4 as well. Not sure.

Feel free to open a PR into my branch if you're motivated, or tell me the values that work well and I'll put it in. We might as well make the scalings work for your data, and hope it generalizes well to data from other people.

**Comment by Eric89GXL** [comment]
Or actually `scalings_plot_raw = dict(..., seeg=1e-5, bio=1e-6, ecog=1e6)` might need to change. I think I meant to make that `1e-6`!

**Comment by Eric89GXL** [comment]
Changing to WIP until @choldgraf is happy :)

**Comment by choldgraf** [comment]
So the defaults assume that the data is stored in volts (for electrical signals) right? I looked through a few of our lab's data and it's actually stored in microvolts, so it's basically already been scaled. That said, 1e6 is probably correct if you're assuming that people are loading it in in volts.

**Comment by choldgraf** [comment]
Actually looking at a bunch of channels, it's often in "hundreds of millivolts", so maybe 1e-4 (also realized I missed the negation in the exponent too in my previous comment). e.g.:

image

**Comment by Eric89GXL** [comment]
Are your data usually recorded from the  as EDF? Maybe it would be
informative to work directly from one of those files. Feel free to ping me
privately if you have one to share non-publicly, or better upload somewhere
if you have a small EDF file we could add to the test dataset. I suspect
we'll want to assume or convert data to units of V to be consistent with
the rest of the package and channel types.

**Comment by Eric89GXL** [comment]
From the hardware, I mean (silly mobile)

**Comment by choldgraf** [comment]
Sometimes they're recorded in EDF, but not always. Our lab gets data from a bunch of different hospitals, each with their own way of doing things (e.g., different recording systems, different storage systems, different data formats). It's actually kind of hard to find data that's totally "raw", but the edf file that I loaded for the pic above was pretty close to that. I think it's fine to expect units in V, so maybe 1e-4 is the way to go.

**Comment by Eric89GXL** [comment]
So what variable needs changing, the raw one? Then good to go from your end?

**Comment by Eric89GXL** [comment]
Scaling fixed, ready for review/merge from my end. If the scalings are off a little bit we can tweak as use cases pop up.

File: mne/io/pick.py [hunk 1]
```diff
@@ -667,25 +674,27 @@ def _check_excludes_includes(chs, info=None, allow_bads=False):
...
 
     Corresponds to EOG, ECG, EMG and BIO
     """
-    return pick_types(info, meg=False, eeg=False, stim=False, eog=True,
-                      ecg=True, emg=True, ref_meg=False, misc=False,
-                      resp=False, chpi=False, exci=False, ias=False,
-                      syst=False, seeg=False, dipole=False, gof=False,
-                      bio=True, include=[], exclude=exclude, selection=None)
+    return pick_types(info, meg=False, eog=True, ecg=True, emg=True, bio=True,
+                      exclude=exclude)
```
Comment (jaeilepp) [inline mne/io/pick.py]: Shouldn't ``ref_meg`` be False?

File: mne/preprocessing/ica.py [hunk 2]
```diff
@@ -298,21 +298,21 @@ def fit(self, inst, picks=None, start=None, stop=None, decim=None,
...
-            Valid keys are 'grad' | 'mag' | 'eeg' | 'eog' | 'ecg'.
+            Valid keys are 'grad', 'mag', 'eeg', 'seeg', 'ecog', 'eog', 'ecg'.
             If reject is None then no rejection is done. Example::
 
                 reject = dict(grad=4000e-13, # T / m (gradiometers)
                               mag=4e-12, # T (magnetometers)
-                              eeg=40e-6, # V (EEG channels)
-                              eog=250e-6 # V (EOG channels)
+                              eeg=40e-6, # uV (EEG channels)
+                              eog=250e-6 # uV (EOG channels)
```
Comment (jaeilepp) [inline mne/preprocessing/ica.py]: Aren't the units in V if 1e-6?

File: doc/whats_new.rst [hunk 3]
```diff
@@ -138,7 +140,7 @@ API
 
     - To unify and extend the behavior of :func:`mne.comupute_raw_covariance` relative to :func:`mne.compute_covariance`, the default parameter ``tstep=0.2`` now discards any epochs at the end of the :class:`mne.io.Raw` instance that are not the full ``tstep`` duration. This will slightly change the computation of :func:`mne.compute_raw_covaraince`, but should only potentially have a big impact if the :class:`mne.io.Raw` instance is short relative to ``tstep`` and the last, too short (now discarded) epoch contained data inconsistent with the epochs that preceded it.
 
-    - The default ``picks=None`` in :func:`mne.io.Raw.filter` nows picks eeg, meg and seeg channels, by `Jean-Remi King`_
+    - The default ``picks=None`` in :func:`mne.io.Raw.filter` nows picks eeg, meg, seeg, and ecog channels, by `Jean-Remi King`_ and `Eric Larson`_
```
Comment (jaeilepp) [inline doc/whats_new.rst]: nows -> now

File: mne/io/pick.py (hunk 1 above)
Comment (Eric89GXL) [inline mne/io/pick.py]: good call

File: mne/preprocessing/ica.py (hunk 2 above)
Comment (Eric89GXL) [inline mne/preprocessing/ica.py]: yeah, I missed this in @kingjr's commits

**Comment by Eric89GXL** [comment]
Thanks @jaeilepp comments addressed

**Comment by jaeilepp** [comment]
LGTM

**Comment by choldgraf** [comment]
Cool! @Eric89GXL maybe we should chat in an issue / PR about figuring out some good testing data etc for ecog development?

**Comment by Eric89GXL** [comment]
Sure, or on Gitter if now is good.

**Comment by choldgraf** [comment]
Hopefully I'm re-submitting this paper sometime between today and
Wednesday, so can I ping you after that? :)

**Comment by Eric89GXL** [comment]
Yeah sounds good

**Comment by choldgraf** [comment]
Is there any input you'd like from me on this PR? Just making sure I'm not holding anything up...

**Comment by Eric89GXL** [comment]
Nope, this one's good to go

File: mne/defaults.py [hunk 4]
```diff
@@ -10,25 +10,26 @@
     color=dict(mag='darkblue', grad='b', eeg='k', eog='k', ecg='m',
                emg='k', ref_meg='steelblue', misc='k', stim='k',
                resp='k', chpi='k', exci='k', ias='k', syst='k',
-               seeg='k', dipole='k', gof='k', bio='k'),
+               seeg='k', dipole='k', gof='k', bio='k', ecog='k'),
```
Comment (jaeilepp) [inline mne/defaults.py]: I feel like we could be more funky with our colors in the future. At least we could have a color scheme where the data channels would be easier to separate.
In following PRs...

File: mne/viz/raw.py [hunk 5]
```diff
@@ -238,12 +239,12 @@ def plot_raw(raw, events=None, duration=10.0, start=0.0, n_channels=20,
         inds += [pick_types(info, meg=t, ref_meg=False, exclude=[])]
         types += [t] * len(inds[-1])
     pick_kwargs = dict(meg=False, ref_meg=False, exclude=[])
-    for t in ['eeg', 'seeg', 'eog', 'ecg', 'emg', 'ref_meg', 'stim', 'resp',
-              'misc', 'chpi', 'syst', 'ias', 'exci', 'bio']:
-        pick_kwargs[t] = True
-        inds += [pick_types(raw.info, **pick_kwargs)]
-        types += [t] * len(inds[-1])
-        pick_kwargs[t] = False
+    for key in _PICK_TYPES_KEYS:
```
Comment (jaeilepp) [inline mne/viz/raw.py]: ``_PICK_TYPES_KEYS`` does not contain ``ref_meg``.

File: mne/defaults.py (hunk 4 above)
Comment (agramfort) [inline mne/defaults.py]: agreed :)

**Comment by agramfort** [comment]
thanks @Eric89GXL !

File: mne/viz/raw.py (hunk 5 above)
Comment (jaeilepp) [inline mne/viz/raw.py]: This was left undone, but I'll make a fix in my PR.

Comment (Eric89GXL) [inline mne/viz/raw.py]: Thanks @jaeilepp

File: mne/defaults.py (hunk 4 above)
Comment (choldgraf) [inline mne/defaults.py]: Isn't matplotlib going to change their default color cycle in the next release?

**Pull request merged**